docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,308 @@
+Assuming that at most `1/3` of validators could be malicious, by having 50% as the minimum value for `top_N` we guarantee that we cannot have a successful incorrect-execution attack on a Top N consumer chain. This is because, a Top N consumer chain with `N >= 50%` would have at least `1/3` correct validators that would be able to censor any incorrect-execution attack.
+Additionally, by having a `N >= 50%` and hence `N > 33%` we provide the ability for validators to `Veto` a `ConsumerAdditionProposal` if they do not desire to validate a consumer chain.
+
+`top_N` can be up to 95% to capture the current case of Replicated Security where we allow the bottom 5% of validators to opt out. There is no reason for `top_N` to be higher than 95%. Smaller chains that belong in the bottom 5% validators can choose to opt in if they want to validate.
+
+So, the fields can take the following values:
+| `is_top_N` | `top_N`|
+| :--- | :---|
+| `true` | `[50, 95]` |
+| `false` | not set |
+
+
+If a proposal has those arguments wrongly set, it should get rejected in [ValidateBasic](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/types/proposal.go#L86).
+
+Note that we could easily distinguish whether a chain is _Top N_ or _Opt In_ by only having the single field `top_N` and if `top_N == 0`, then we’re in the _Opt In_ case, although for clarity, we introduce both fields.
+
+In a future version of PSS, we intend to introduce a way to modify the parameters of a consumer chain, e.g, a chain that is _Opt In_ to become _Top N_, etc.
+
+#### State & Query
+We augment the provider module’s state to keep track of the `is_top_N` and `top_N` values for each consumer chain. The key to store this information would be:
+
+```
+ChainIdWithLenKey(IsTopNBytePrefix, chainID)
+```
+reusing [`ChainIdWithLenKey`](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/types/keys.go#L418).
+
+Then in the [keeper](https://github.com/cosmos/interchain-security/blob/v3.3.0/x/ccv/provider/keeper/keeper.go) we introduce methods as follows:
+```
+func (k Keeper) SetIsTopN(ctx sdk.Context, chainID string, isTopN bool) error
+func (k Keeper) IsTopN(ctx sdk.Context, chainID string) bool
+
+// returns the N if Top N chain, otherwise an error
+func (k Keeper) GetTopN(ctx sdk.Context, chainID string) (uint8, error)
+```
+
+In a future version of PSS, we intend to extend the `interchain-security-pd query provider list-consumer-chains` query to also return information on whether a consumer chain is an Opt In or a Top N chain and with what N. This way, block explorers like [Mintscan](https://www.mintscan.io/) would show for a chain “This chain is secured by N% of Cosmos Hub.”
+
+### How do validators opt in?
+A validator can opt in by sending a new type of message that we introduce in [tx.proto](https://github.com/cosmos/interchain-security/blob/v3.3.0/proto/interchain_security/ccv/provider/v1/tx.proto#L1).
+```
+message MsgOptIn {
```

Comment:
I would keep the messages separate for now, and add extra conveniences like combined messages once the system is built.

Reply:
I agree to not complicate now, but we should take these UX decisions before releasing PSS as afterwards it will require more comms to notify users that we changed the API

Reply:
Added consumer key in the message.